### PR TITLE
Add support to extract the session details based on the tenant

### LINKED ISSUE #13465: SLO between tenants does not work when IDP name is different
**Describe the issue:**
To achieve SSO between tenants, each tenant can be exposed through a separate hostname as mentioned in the blog post[1]. The SSO between tenants is working as expected with this approach however when trying to log out there are a few issues being faced.

Assume the following scenario for these issues.

diagram

Each of the tenants is accessed through a different hostname as shown in the above diagram. The flow is as follows:

user logs in to app1, which is in the app1.com tenant, it has the federated authentication configured for the idp.com tenant, so the user is directed to the idp.com tenant to login. the user enters the credentials and logins to the app1.

user also logs into app2, which is in the app2.com tenant, it also has the federated authentication configured for the idp.com tenant, so the user is directed to the idp.com tenant to log in again. Since there is already a session at the idp.com tenant for the user, they would be logged in automatically(SSO).

The above scenarios work fine without any issues.

**How to reproduce:**

1. Now user initiates a logout from app1, the oidc logout is sent to the app1.com tenant
2. app1.com tenant in turn sends a logout request to the federated IDP - idp.com tenant. 
3. At the federated IDP - idp.com tenant, it identifies that there is another application that needs to be logged out and issues the SLO request to the app2.com tenant[2]

There are two issues in this case:

- At the federated IDP, the SLO logout request to the app2.com tenant is issued with the wrong issuer, the issuer being used is the IDP entity id alias configured in the SP for app1.com. Ideally, the SLO should be issued for the SP with the IDP entity alias configured in its service provider. (this occurs because we are getting the id entity id from the current SP that is stored in a threadlocal variable[3][4])

- At the app2.com tenant, when trying to process the SLO request an error occurs saying that the IDP cannot be found. (if the same IDP name is configured in the app1.com and app2.com tenants the flow works as expected, in the IDN_FED_AUTH_SESSION_MAPPING table we update the entry created for the first app1.com tenant login[5]). 

```
Caused by: java.lang.NullPointerExceptionCaused by: java.lang.NullPointerException
at org.wso2.carbon.identity.application.authenticator.samlsso.logout.util.SAMLLogoutUtil.getFederatedIdPConfigs(SAMLLogoutUtil.java:155)
at org.wso2.carbon.identity.application.authenticator.samlsso.logout.processor.SAMLLogoutRequestProcessor.populateContextWithSessionDetails(SAMLLogoutRequestProcessor.java:218)
at org.wso2.carbon.identity.application.authenticator.samlsso.logout.processor.SAMLLogoutRequestProcessor.process(SAMLLogoutRequestProcessor.java:120)
at org.wso2.carbon.identity.application.authenticator.samlsso.logout.processor.SAMLLogoutRequestProcessor.process(SAMLLogoutRequestProcessor.java:75)
at org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityProcessCoordinator.process(IdentityProcessCoordinator.java:41)
at org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityServlet.process(IdentityServlet.java:84) ... 39 more
```
[1]http://xacmlinfo.org/2015/11/24/service-provider-grouping-with-wso2-identity-server/
[2]https://is.docs.wso2.com/en/5.10.0/learn/handling-saml-single-logout-requests-from-federated-identity-providers/
[3]https://github.com/wso2-support/identity-inbound-auth-saml/blob/a352e68d4d0f9ea1fcd914107c121f1085c9d94a/components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java#L559
[4]https://github.com/wso2-support/identity-inbound-auth-saml/blob/a352e68d4d0f9ea1fcd914107c121f1085c9d94a/components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java#L617
[5]https://github.com/wso2-support/carbon-identity-framework/blob/71bbf543ec1b37b992faa79b885555ec58728137/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/UserSessionStore.java#L885-L907

**Environment information** **:**
 - Product Version: [IS 5.10.0]

### PR DESCRIPTION
**Pull request by ShanChathusanda93**
## Changes Proposed in this PR
* Fixes : https://github.com/wso2/product-is/issues/13465
* Added the support to extract the session details by using the tenant id.
* When extracting the IDP details, the search in the IDN_FED_AUTH_SESSION_MAPPING table will be happenning with the tenant id.
* Upgraded the kernel version to `4.7.0-beta` to fix the pax logging issue when upgrading the framework version since the pax logging is removed by [2].
* Upgraded the opensaml version to `3.3.1.wso2v6` since it is the latest version used[3].
* Adding unit test cases to test the flow as well as to cover the minimum tests coverage.

## Related PRs
* Framework fix which was done to introduce the tenant id column and storing the session details with the tenant[1].

## When should this PR be merged
* This PR needs to be merged after the framework PR[1] is merged and need to update the framework version also.

[1] https://github.com/wso2/carbon-identity-framework/pull/4034
[2] https://github.com/wso2/carbon-identity-framework/pull/4008
[3] https://github.com/wso2/orbit/tree/master/opensaml/3.3.1.wso2v6

### CODE REVIEW

**Review by nilasini — APPROVED** [review]

File: components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/dao/SessionInfoDAO.java [hunk 1]
```diff
@@ -75,4 +75,38 @@ public Map<String, String> getSessionDetails(String samlIndex) throws SAMLLogout
...
+            preparedStatement.setString(1, samlIndex);
+            preparedStatement.setInt(2, tenantId);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                Map<String, String> sessionDetails = new HashMap<>();
+                if (resultSet.next()) {
+                    sessionDetails.put(SESSION_ID, resultSet.getString("SESSION_ID"));
+                    sessionDetails.put(IDP_NAME, resultSet.getString("IDP_NAME"));
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Retrieved session id: " + resultSet.getString("SESSION_ID") +
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/dao/SessionInfoDAO.java]: what would happen here if the result set was empty?

Comment (ShanChathusanda93) [reply components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/dao/SessionInfoDAO.java]: fixed

**Review by janakamarasena — APPROVED** [review]
